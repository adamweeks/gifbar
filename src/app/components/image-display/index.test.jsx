import React from 'react';
import renderer from 'react-test-renderer';

import ImageDisplay from '.';

describe(`ImageDisplay component`, () => {
  const handleClick = jest.fn();
  const onCopy = jest.fn();
  const onOpenModal = jest.fn();
  const image = {
    id: 1,
    displayUrl: `http://mycoolserver.info/kid-n-play.gif`,
    fullSizedImageFileSize: `5 Mb`,
    imageSizes: {
      smallSize: {
        height: 50
      }
    }
  };
  const component = renderer.create(
    <ImageDisplay
      handleClick={handleClick}
      image={image}
      onCopy={onCopy}
      onOpenModal={onOpenModal}
    />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});
