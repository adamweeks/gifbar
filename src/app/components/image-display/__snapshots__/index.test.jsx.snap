// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDisplay component renders properly 1`] = `
<div
  className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginRight-61z16t rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  style={
    Object {
      "marginBottom": "3px",
      "marginTop": "3px",
    }
  }
>
  <div
    className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(204,204,204,1)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "contain",
        "height": "50px",
        "width": "200px",
      }
    }
  />
</div>
`;
